tests/Endatix.Core.Tests/Entities/SubmissionConstructorTests.cs

```diff
@@ -0,0 +1,79 @@
+using Endatix.Core.Entities;
+using static Endatix.Core.Tests.ErrorMessages;
+using static Endatix.Core.Tests.ErrorType;
+
+namespace Endatix.Core.Tests.Entities;
+
+public class SubmissionConstructorTests
+{
+    [Fact]
+    public void Constructor_NullJsonData_ThrowsArgumentNullException()
+    {
+        // Act
+        var action = () => new Submission(null);
+
+        // Assert
+        action.Should().Throw<ArgumentNullException>()
+            .WithMessage(GetErrorMessage(nameof(Submission.JsonData), Null));
+    }
+
+    [Fact]
+    public void Constructor_EmptyJsonData_ThrowsArgumentException()
+    {
+        // Act
+        var action = () => new Submission(string.Empty);
+
+        // Assert
+        action.Should().Throw<ArgumentException>()
+            .WithMessage(GetErrorMessage(nameof(Submission.JsonData), Empty));
+    }
+
+    [Fact]
+    public void Constructor_NegativeFormDefinitionId_ThrowsArgumentException()
+    {
+        // Arrange
+        const string jsonData = SampleData.SUBMISSION_JSON_DATA_1;
+        const long invalidFormDefinitionId = -1;
+
+        // Act
+        var action = () => new Submission(jsonData, invalidFormDefinitionId);
+
+        // Assert
+        action.Should().Throw<ArgumentException>()
+            .WithMessage(GetErrorMessage(nameof(Submission.FormDefinitionId), ZeroOrNegative));
+    }
+
+    [Fact]
+    public void Constructor_ValidInput_SetsPropertiesCorrectly()
+    {
+        // Arrange
+        const string jsonData = SampleData.SUBMISSION_JSON_DATA_1;
+        const long formDefinitionId = 123;
+
+        // Act
+        var submission = new Submission(jsonData, formDefinitionId, isComplete: false, currentPage: 2, metadata: "Test");
+
+        // Assert
+        submission.Should().NotBeNull();
+        submission.JsonData.Should().Be(jsonData);
+        submission.FormDefinitionId.Should().Be(formDefinitionId);
+        submission.IsComplete.Should().BeFalse();
+        submission.CurrentPage.Should().Be(2);
+        submission.Metadata.Should().Be("Test");
+    }
+
+    [Fact]
+    public void Constructor_CompleteSubmission_SetsCompletedAt()
+    {
+        // Arrange
+        const string jsonData = SampleData.SUBMISSION_JSON_DATA_1;
+
+        // Act
+        var submission = new Submission(jsonData, isComplete: true);
+
+        // Assert
+        submission.Should().NotBeNull();
+        submission.IsComplete.Should().BeTrue();
+        submission.CompletedAt.Should().NotBeNull();
```

Comment:
I like that you have improved the coverage by adding more tests. Just an idea for work in another task. We should add TimeProvider abstraction for testing things with DataTime easier. Should be an easy one (https://blog.nimblepros.com/blogs/finally-an-abstraction-for-time-in-net/). If you agree, I can create a task

- [ ] refactor: abstract time provider

Reply:
Yes, this is the way to properly test time data. Please add a task. It should:
- introduce the provider where necessary
- change the exiting tests to utilize it
- update [Initial Prompt Template](https://drive.google.com/file/d/1PPy3FaEQ_TfRTxWhCpH5zNFwicK6qrA2/view?usp=drive_link) if needed to adjust tests creation.